code/modules/antagonists/paradox_clone/datum_paradox_clone.dm

```diff
@@ -0,0 +1,177 @@
+	"Romeo", "Sierra", "Tango", "Uniform", "Victor", "Whiskey", "X-ray",
+	"Yankee", "Zulu")) // just a copy of GLOB.photenic_alphabet but for paradox clones ids which removes
+
+GLOBAL_LIST_INIT(paradox_clones, list())
+
+/datum/antagonist/paradox_clone
+	name = "Paradox Clone"
+	roundend_category = "paradox clones"
+	job_rank = ROLE_PARADOX_CLONE
+	special_role = SPECIAL_ROLE_PARADOX_CLONE
+	antag_hud_name = "hudparadoxclone"
+	antag_hud_type = ANTAG_HUD_PARADOX_CLONE // they see each other hud
+	clown_gain_text = "<span class='notice'>You are from alternative universe, where clowns are actually quite dextrous. This allows you to wield weapons without harming yourself.</span>"
+	clown_removal_text = "<span class='danger'>Your connection with your native universe fades, and with it goes your dexterity.</span>"
+	wiki_page_name = "Paradox_Clone"
+	var/paradox_id = "Paradox"
+	var/real_id
+	var/static/list/paradox_powers = list(
+		/datum/spell/paradox_spell/click_target/gaze,
+		/datum/spell/touch/paradox_spell/space_distortion,
+		/datum/spell/paradox_spell/self/digital_supersede,
+		/datum/spell/paradox_spell/click_target/mind_interference,
+		/datum/spell/paradox_spell/self/illusion,
+		/datum/spell/paradox_spell/click_target/energy_exchange,
+		/datum/spell/touch/paradox_spell/microcircuit_disorder,
+		/datum/spell/paradox_spell/self/intangibility,
+		/datum/spell/touch/paradox_spell/suppression,
+		/datum/spell/paradox_spell/aoe/hud_malfunction
+	)
+	// "replace" is issued if there is a objective to kill n replace and "United Bonds" when need to protect original.
+	var/mob/living/carbon/human/original
+	var/list/current_powers = list()
+	give_objectives = FALSE
+	silent = TRUE
+
+	blurb_text_color = COLOR_GRAY
+	blurb_text_outline_width = 2
+
+	blurb_r = 46
+	blurb_g = 46
+	blurb_b = 46
+	blurb_a = 0.40
+
+/datum/antagonist/paradox_clone/on_gain()
+	var/mob/living/carbon/human/H = owner.current
+	GLOB.paradox_clones += owner
+	var/prefix = pick("Experiment", "Paradox", "Agent", "Subject") // There is some hint that some paradoxical clones are victims of an anomaly, either experiments by scientists, or specially sent agents.
+	if(length(GLOB.possible_paradox_clone_ids))
+		var/id = pick(GLOB.possible_paradox_clone_ids)
+		real_id = id
+		GLOB.possible_paradox_clone_ids -= id
+		paradox_id = "[prefix] [id]"
+	else
+		paradox_id = "[prefix] [rand(1,999)]"
+
+	for(var/mob/living/carbon/human/clone in GLOB.human_list)
+		if(clone != H && is_paradox_clone(clone))
+			to_chat("<span class='danger'>You have a feeling, that something familiar and native appeared in this dark universe...</span>")
+
+	owner.special_role = special_role
+	apply_innate_effects()
+
+	if(is_banned(owner.current) && replace_banned)
+		INVOKE_ASYNC(src, PROC_REF(replace_banned_player))
+	owner.current.create_log(MISC_LOG, "[owner.current] was made into \an [special_role]")
+	return TRUE // override cus shows up empty red box menu... not cool!
```

Comment:
All antag datums really should be calling parent `on_gain()` to avoid duplicated code and to do the necessary things like call `greet()` and such. What is the red box issue mentioned here?

Reply:
this.

to_chat(owner.current, chat_box_red(messages.Join("<br>")))

and this red box are empty cus objectives are gaining outside of "give_objectives proc"